docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,472 @@
+       >
+       >When the `isDouble` variable of `TiDAGRequest` is true, the `CoprocessorRDD` is called `FetchHandleRDD`.
+
+    3. information about the used index is hard to know
+
+       In the physical plan explanation, only the name of the index we use is shown, not the columns that make up the index.
+
+   ```SQL
+   CREATE TABLE `t2` (
+     `a` BIGINT(20) UNSIGNED  NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`,`b`)
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```SQL
+   SELECT * FROM t2 where a<1 or b>'bb'"
+   ```
+
+   ```text
+   == Physical Plan ==
+   *(1) ColumnarToRow
+   +- TiSpark RegionTaskExec{downgradeThreshold=1000000000,downgradeFilter=[[[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]]]
+      +- RowToColumnar
+         +- TiKV FetchHandleRDD{[table: t2] IndexScan[Index: primary] , Columns: a@UNSIGNED LONG, b@VARCHAR(255), c@VARCHAR(255), Downgrade Filter: [[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]], PushDown Filter: [[a@UNSIGNED LONG LESS_THAN 1] OR [b@VARCHAR(255) GREATER_THAN "bb"]], KeyRange: [([t\200\000\000\000\000\000\003\374_i\200\000\000\000\000\000\000\001\000], [t\200\000\000\000\000\000\003\374_i\200\000\000\000\000\000\000\001\372])], Aggregates: , startTs: 433735624236728323}
+   ```
+
+3. expression(s) that constitutes the scanning range is hard to know
+
+   The expression(s) for building ranges are not shown explicitly, which makes it difficult to know which expression(s) are used to build range. As shown below, for the query expression `a>0` is pushed down, but the pushed-down information is only implicitly given inside the `KeyRange`, which is not convenient for users to understand.
+
+   ```sql
+   CREATE TABLE `t3` (
+     `a` BIGINT(20) NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) clustered
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```sql
+   SELECT a FROM t3 where a>0
+   ```
+
+   ```text
+   == Physical Plan ==
+   *(1) ColumnarToRow
+   +- TiKV CoprocessorRDD{[table: t3] TableScan, Columns: a@LONG, KeyRange: [([t\200\000\000\000\000\000\005/_r\200\000\000\000\000\000\000\001], [t\200\000\000\000\000\000\005/_s\000\000\000\000\000\000\000\000])], startTs: 434063568626778113}
+   ```
+
+4. confusing naming of the operator
+
+   In TiDB, a process named `Scan` means scanning a piece of data, not aggregating and computing the result of the scan. But in TiSpark, a process named scan means aggregating and computing the results of the scan. This may confuse users who switch from TiDB to TiSpark.
+
+   The selection expression pushed down to TiKV/COP in TiDB is called `Selection`, while the selection expression is called `Pushdown Filter` in TiSpark. This may also confuse users who switch from TiDB to TiSpark.
+
+## Detailed Design
+
+### appear a filter that should not appear
```

Comment:
Maybe it is inappropriate to make a problem as a title in Detailed Design?

Reply:
I think it is better for us to understand how to resolve the problem